### DISCUSSION TITLE
PHP formatting on save difference than with VS Code

### DISCUSSION THREAD

**Discussion by kjvdven** [zed-industries/zed#19879] [Feature Requests]
I have a WordPress project using Roots 9 from Sage. This uses a `phpcs.xml` to setup PHP Code Sniffer.

Now I want Zed to format my code just like VS Code does, so that people can easily choose their editor. I set up both editors to use PHP Actor as LSP. 

Zed with PHP Actor notices it's a Code Sniffer project and makes an PHP Actor settings file, and I do get proper warnings. 

As example: I have a rule set to change the indenting to 2 spaces.

```xml
  <rule ref="Generic.WhiteSpace.ScopeIndent">
      <properties>
          <property name="indent" value="2"/>
      </properties>
  </rule>
```

1. If the spacing is wrong I do get a warning from Zed, saying the indenting is wrong
2. If I right click, I can format with Code Sniffer --> I get 2 spaces
3. If I save the file in Zed, the file is formatted --> I get 4 spaces
4. If I save the file in VS Code the file is formatted --> I get 2 spaces

Can I "correct" this behavior in the Zed config? 
It feels like when formatting the Code Sniffer rules are ignored by PHP Actor in Zed.

```jsonc
    "PHP": {
      "language_servers": ["phpactor", "!intelephense"]
      // "formatter": {}
    },
```

**Comment by bajrangCoder** [comment]
I guess zed try to format using prettier. Try changing formatter:

```json
{
  "formatter": "language_server"
}

```

[Zed docs (formatter)](https://zed.dev/docs/configuring-zed?highlight=Formatter#formatter)

**Reply by kjvdven** [reply]
The configuration I'm trying out now:

```jsonc
    "PHP": {
      "language_servers": ["phpactor", "!intelephense", "..."],
      "formatter": "prettier", // is this necessary? 
      "code_actions_on_format": { 
        "source.fixAll.phpactor": true // does this do anything?
      }
    },
```

It looks like something is formatted 🤔

**Reply by bajrangCoder** [reply]
> ```
> "formatter": "prettier", // is this necessary? 
> ```

Try by using `language-server` instead of prettier , as you want to format it using php actor.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
